From d41e452e37d2777b797c5a0cb1a1f66e90900c46 Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Thu, 4 Dec 2014 15:41:27 -0600
Subject: [PATCH] sw1705-condor_ce_reconfig_error_msg

Add a user-friendly error message when reconfig fails in
condor_ce_config_generator (SOFTWARE-1705)
---
 src/condor_ce_config_generator | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/condor_ce_config_generator b/src/condor_ce_config_generator
index 6dd526a..a144cad 100755
--- a/src/condor_ce_config_generator
+++ b/src/condor_ce_config_generator
@@ -361,11 +361,15 @@ def main():
             if not opts.verbose:
                 proc_outerr_fd = logger.handlers[0].stream.fileno()
 
             retcode = subprocess.call(cmd, stdout=proc_outerr_fd, stderr=proc_outerr_fd)
             if retcode != 0:
-                raise subprocess.CalledProcessError(retcode, cmd)
+                logger.warning("condor_ce_reconfig failed.  This may be"
+                               " because the condor-ce daemons aren't"
+                               " running.  Rerun with --no-reconfig"
+                               " to supress this message.")
+                sys.exit(1)
     else:
         logger.info("Final config:")
         logger.info(config)
         logger.info("Dry-run enabled; not actually writing configuration file.")
 
-- 
2.0.1

